scripts/dts/new_extract.py

```diff
@@ -60,35 +60,47 @@ def main():
             print("#define DT_SRAM_BASE_ADDRESS\t"
                       + hex(edt.sram_dev.regs[0].addr),
                   file=out)
+            print("#define DT_SRAM_SIZE\t",
+                      + edt.sram_dev.regs[0].size // 1024,
+                      sep="", file=out)
 
         if edt.ccm_dev:
             print("#define DT_CCM_BASE_ADDRESS\t"
                       + hex(edt.ccm_dev.regs[0].addr),
                   file=out)
+            print("#define DT_CCM_SIZE\t",
+                      + edt.ccm_dev.regs[0].size // 1024,
+                      sep="", file=out)
 
 
 def write_regs(dev, out):
     for reg in dev.regs:
-        print("#define {}\t0x{:x}".format(reg_ident(reg), reg.addr), file=out)
+        print("#define {}\t0x{:x}".format(reg_ident(reg, "BASE_ADDRESS"), reg.addr), file=out)
+        if reg.size:
+            print("#define {}\t{:d}".format(reg_ident(reg, "SIZE"), reg.size), file=out)
 
 
 def write_aliases(dev, out):
     for reg in dev.regs:
-        ident = reg_ident(reg)
+        ident = reg_ident(reg, "BASE_ADDRESS")
+        ident_size = reg_ident(reg, "SIZE")
         for alias in reg_aliases(reg):
             # Avoid writing aliases that overlap with the base identifier for
             # the register
             if alias != ident:
                 print("#define {}\t{}".format(alias, ident), file=out)
+                if reg.size:
+                    alias = alias.replace("BASE_ADDRESS", "SIZE")
```

Comment:
Could these be returned from `reg_aliases()` instead, to avoid the rewriting?

`reg_alias()` is supposed to return all the aliases.